service/history/execution/mutable_state_task_refresher.go

```diff
@@ -218,10 +218,14 @@ func (r *mutableStateTaskRefresherImpl) refreshTasksForWorkflowClose(
 ) error {
 
 	executionInfo := mutableState.GetExecutionInfo()
-
 	if executionInfo.CloseStatus != persistence.WorkflowCloseStatusNone {
+		closeEvent, err := mutableState.GetCloseEvent(ctx)
+		if err != nil {
+			return err
+		}
 		return taskGenerator.GenerateWorkflowCloseTasks(
 			now,
+			closeEvent,
```

Comment:
it's possible that the closeEvent returned is nil even when error is nil right? then it will cause a NPE in taskGenerator?

Reply:
Will return error when close event is nil